test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,433 @@
+    std::vector<std::string> messages =
+        absl::StrSplit(data_message, "data:", absl::SkipWhitespace());
+    for (auto message : messages) {
+      if (absl::StrContains(getStreamField(message, "type"), "HystrixCommand")) {
+        std::string cluster_name(getStringStreamField(message, "name"));
+        cluster_message_map[cluster_name] = message;
+      }
+    }
+    return cluster_message_map;
+  }
+
+  TestRandomGenerator rand_;
+  uint64_t window_size_ = rand_.random() % 10 + 5; // Arbitrary reasonable number.
+  const std::string cluster1_name_{"test_cluster1"};
+  ClusterTestInfo cluster1_{cluster1_name_};
+
+  // second cluster for "end and remove cluaster" tests
+  const std::string cluster2_name_{"test_cluster2"};
+  ClusterTestInfo cluster2_{cluster2_name_};
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  NiceMock<Server::MockInstance> server_;
+  Upstream::ClusterManager::ClusterInfoMap cluster_map_;
+
+  std::unique_ptr<HystrixSink> sink_;
+  NiceMock<Stats::MockSource> source_;
+  NiceMock<Upstream::MockClusterManager> cluster_manager_;
+};
+
+TEST_F(HystrixSinkTest, EmptyFlush) {
+  InSequence s;
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+  // Register callback to sink.
+  sink_->registerConnection(&callbacks_);
+  sink_->flush(source_);
+  std::string data_message = TestUtility::bufferToString(buffer);
+  validateResults(data_message, 0, 0, 0, 0, 0, window_size_);
+}
+
+TEST_F(HystrixSinkTest, BasicFlow) {
+  InSequence s;
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+  // Register callback to sink.
+  sink_->registerConnection(&callbacks_);
+
+  // Only success traffic, check randomly increasing traffic
+  // Later in the test we'll "shortcut" by constant traffic
+  uint64_t traffic_counter = 0;
+  ;
+  sink_->flush(source_); // init window with 0
+  for (uint64_t i = 0; i < (window_size_ - 1); i++) {
+    buffer.drain(buffer.length());
+    traffic_counter += rand_.random() % 1000;
+    ON_CALL(cluster1_.success_counter_, value()).WillByDefault(Return(traffic_counter));
+    sink_->flush(source_);
+  }
+
+  EXPECT_EQ(getStreamField(TestUtility::bufferToString(buffer), "rollingCountSuccess"),
+            std::to_string(traffic_counter));
+  EXPECT_EQ(getStreamField(TestUtility::bufferToString(buffer), "requestCount"),
+            std::to_string(traffic_counter));
+  EXPECT_EQ(getStreamField(TestUtility::bufferToString(buffer), "errorCount"), "0");
+  EXPECT_EQ(getStreamField(TestUtility::bufferToString(buffer), "errorPercentage"), "0");
+
+  // Check mixed traffic
+  // Values are unimportant - they represent traffic statistics, and for the purpose of the test any
+  // arbitrary number will do. Only restriction is that errors >= timeouts, since in Envoy timeouts
+  // are counted as errors and therefore the code that prepares the stream for the dashboard reduces
+  // number of timeouts from total number of errors.
```

Comment:
nit: s/reduces/deducts the/ 

You could use either "reduce ... by" or "deducts ... from".